unofficial/c511027787.lua

```diff
@@ -0,0 +1,102 @@
+--ＲＲ－レディネス (アニメ)
+--Raidraptor - Readiness (Anime)
+--Made by When
+local s,id=GetID()
+function s.initial_effect(c)
+    --Activate (if you would take battle damage)
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+	e1:SetType(EFFECT_TYPE_ACTIVATE)
+	e1:SetCode(EVENT_PRE_DAMAGE_CALCULATE)
+	e1:SetCondition(s.condition1)
+	e1:SetOperation(s.activate1)
+	c:RegisterEffect(e1)
+	--Activate (if you would take effect damage)
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(id,0))
+	e2:SetProperty(EFFECT_FLAG_DAMAGE_STEP)
+	e2:SetType(EFFECT_TYPE_ACTIVATE)
+	e2:SetCode(EVENT_CHAINING)
+	e2:SetCondition(s.condition2)
+	e2:SetOperation(s.activate2)
+	c:RegisterEffect(e2)
+	--Banish From GY
+	local e3=Effect.CreateEffect(c)
+	e3:SetDescription(aux.Stringid(id,1))
+	e3:SetProperty(EFFECT_FLAG_DAMAGE_STEP)
+	e3:SetType(EFFECT_TYPE_QUICK_O)
+	e3:SetCode(EVENT_CHAINING)
+	e3:SetRange(LOCATION_GRAVE)
+	e3:SetCost(aux.bfgcost)
+	e3:SetCondition(s.damcon)
+	e3:SetOperation(s.damop)
+	c:RegisterEffect(e3)
+end
+function s.condition2(e,tp,eg,ep,ev,re,r,rp)
+	return aux.damcon1(e,tp,eg,ep,ev,re,r,rp) 
+end
+function s.activate2(e,tp,eg,ep,ev,re,r,rp)
+	local cid=Duel.GetChainInfo(ev,CHAININFO_CHAIN_ID)
+	local e1=Effect.CreateEffect(e:GetHandler())
+	e1:SetType(EFFECT_TYPE_FIELD)
+	e1:SetCode(EFFECT_CHANGE_DAMAGE)
+	e1:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e1:SetTargetRange(1,0)
+	e1:SetLabel(cid)
+	e1:SetValue(s.refval)
+	e1:SetReset(RESET_CHAIN)
+	Duel.RegisterEffect(e1,tp)
+end
+function s.damcon(e,tp,eg,ep,ev,re,r,rp)
+	return aux.damcon1(e,tp,eg,ep,ev,re,r,rp) 
+end
+function s.damop(e,tp,eg,ep,ev,re,r,rp)
+	local cid=Duel.GetChainInfo(ev,CHAININFO_CHAIN_ID)
+	--you take no effect damage
+	local e1=Effect.CreateEffect(e:GetHandler())
+	e1:SetType(EFFECT_TYPE_FIELD)
+	e1:SetCode(EFFECT_CHANGE_DAMAGE)
+	e1:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e1:SetTargetRange(1,0)
+	e1:SetLabel(cid)
+	e1:SetValue(s.refval)
+	e1:SetReset(RESET_CHAIN)
+	Duel.RegisterEffect(e1,tp)
+	--your LP to 10
+	Duel.SetLP(tp,10)
+	--No Battle Damage for the rest of the turn
+	local e2=Effect.CreateEffect(e:GetHandler())
+	e2:SetType(EFFECT_TYPE_FIELD)
+	e2:SetCode(EFFECT_AVOID_BATTLE_DAMAGE)
+	e2:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e2:SetTargetRange(1,0)
+	e2:SetValue(1)
+	e2:SetReset(RESET_PHASE+PHASE_END)
+	Duel.RegisterEffect(e2,tp)
+	local e3=Effect.CreateEffect(e:GetHandler())
+	e3:SetProperty(EFFECT_FLAG_PLAYER_TARGET+EFFECT_FLAG_CLIENT_HINT+EFFECT_FLAG_OATH)

```

Comment:
Not sure which card you copied this effect from, but it is not complete and wrong.
Not complete: This effect was used to add a client hint to the activator to notify both players that the activator is not taking Battle Damage for the rest of the turn, you can tell this from `EFFECT_FLAG_CLIENT_HINT` flag. And it's currently missing the SetDescription that contains the actual client hint message.
Wrong: It should not have EFFECT_FLAG_OATH here, this flag is often used in cost for things like `If you **activate** this effect, you cannot do X for the rest of the turn`. For a restriction like this, if you negate the effect's **activation**, the restriction will no longer apply to you, and the flag `EFFECT_FLAG_OATH` is used to handle this interaction automatically. And clearly this is not relevant to Readiness.

Additionally, to add the player hint message, you could just add the flag `EFFECT_FLAG_CLIENT_HINT` to the e2 above, and add the corresponding SetDescription to it.